core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -920,6 +920,7 @@ class KafkaController(val config: KafkaConfig,
            * triggered by the admin client
            */
           OfflinePartitionLeaderElectionStrategy(allowUnclean = electionTrigger == AdminClientTriggered)
+        case ElectionType.DESIGNATED => throw new RuntimeException("Designated election not supported by KafkaController")
```

Comment:
Something like "in ZK mode" would be more helpful. Also, what is the error passed back to the user?

Why is this not UnsupportedVersionException rather than a generic runtime error?